securedrop_client/gui/actions.py

```diff
@@ -192,3 +195,62 @@ def _on_triggered(self) -> None:
                 self._export_device, "conversation.txt", str(file_path)
             )
             dialog.exec()
+
+
+class ExportConversationAction(QAction):  # pragma: nocover
+    def __init__(
+        self,
+        parent: QMenu,
+        controller: Controller,
+        source: Source,
+        export_service: Optional[export.Service] = None,
+    ) -> None:
+        """
+        Allows export of a conversation transcript.
+        """
+        text = _("Export Conversation Transcript")
+
+        super().__init__(text, parent)
+
+        self.controller = controller
+        self._source = source
+
+        if export_service is None:
+            # Note that injecting an export service that runs in a separate
+            # thread is greatly encouraged! But it is optional because strictly
+            # speaking it is not a dependency of this FileWidget.
```

Comment:
Thread management is the reason why the `export_service` needs to be injected. The export service must run in a thread separate from the main thread so that it doesn't block the UI. (Starting `sd-devices` in particular is a long-running  task by UI interaction standards.)

Additionally, because we are not managing concurrent access to the `sd-devices` service (and have no way to relate the API responses to specific queries), there should be only one instance of it.

I was working on reducing the footprint of that cascade of injections by setting up `export.getService() # singleton` when we decided to postpone that effort and focus on shipping this feature ASAP.

What/when are you suggesting we do?

Reply:
Also: you've used different patterns to indicate that a path is meant for testing (e.g. `self.thing = thing or Thing()`. Do you think any of those may make things clearer in this case?

Subsidiary question: what comment would you suggest that would make things clearer to you?

Reply:
I'm suggesting:
- a comment with less ambiguity than the current one (s/"greatly encouraged"/more specific usage directives, as in your comments here)
- the possibility of code changes that avoid passing around a dependency on export_service outside of where it is used.

Since you're more familiar with the codebase than I am, if you feel that the second point is arduous, unfeasible, or will be addressed in separate work, that may be context that I am missing, and is why I asked it as a more open-ended question rather than requesting changes.

Reply:
Creating a way to acquire an `export_service` instance where it is needed (e.g. `export.getService()`) is one way to avoid passing it down the component tree. (That's what's commonly done with loggers for example.)

If we do that, then the need for the code and comment goes away. Do you want me to rescue that change set and apply it here for evaluation?

Reply:
Yup, I think that's a reasonable approach and I noticed that was part of #1615. If it's easy to rescue + apply, then sure :)

Reply:
Done! :point_down: